docx/parts/document.py

```diff
@@ -126,7 +126,15 @@ def related_hdrftr_body(self, rId):
         Return the |HeaderFooterBody| object corresponding to the related
         part identified by *rId*.
         """
-        raise NotImplementedError
+        part = self.get_related_part(rId)
+        return part.body
+
+    def get_related_part(self, rId):
+        """ HACK this isn't strictly necessary
+        just adding it because it seems much easier to mock than
+        self.rels.related_parts
+        """
+        return self.rels.related_parts[rId]
```

Comment:
Let's keep this all in one piece. Here's an example of how I mocked `related_parts` in python-pptx. `.related_parts` is available directly on the part object, no need to go out to .rels to get it :)

```python
    # in pptx.parts.presentation ------
    def related_slide(self, rId):
        """
        Return the |Slide| object for the related |SlidePart| corresponding
        to relationship key *rId*.
        """
        return self.related_parts[rId].slide

    # the rest of this is in tests.parts.test_presentation ------
    def it_provides_access_to_a_related_slide(self, slide_fixture):
        prs_part, rId, slide_ = slide_fixture
        slide = prs_part.related_slide(rId)
        prs_part.related_parts.__getitem__.assert_called_once_with(rId)
        assert slide is slide_

    @pytest.fixture
    def slide_fixture(self, slide_, related_parts_prop_):
        prs_part = PresentationPart(None, None, None, None)
        rId = 'rId42'
        related_parts_ = related_parts_prop_.return_value
        related_parts_.__getitem__.return_value.slide = slide_
        return prs_part, rId, slide_

    @pytest.fixture
    def related_parts_prop_(self, request):
        return property_mock(request, PresentationPart, 'related_parts')
```